libs/common/src/key-management/crypto/services/encrypt.service.implementation.ts

```diff
@@ -43,22 +43,79 @@ export class EncryptServiceImplementation implements EncryptService {
       return Promise.resolve(null);
     }
 
-    let plainBuf: Uint8Array;
     if (typeof plainValue === "string") {
-      plainBuf = Utils.fromUtf8ToArray(plainValue);
+      return this.encryptUint8Array(Utils.fromUtf8ToArray(plainValue), key);
     } else {
-      plainBuf = plainValue;
+      return this.encryptUint8Array(plainValue, key);
+    }
+  }
+
+  async wrapDecapsulationKey(
+    privateKeyPkcs8: Uint8Array,
+    wrappingKey: SymmetricCryptoKey,
+  ): Promise<EncString> {
+    if (privateKeyPkcs8 == null) {
+      throw new Error("No private key provided for encapsulation.");
+    }
+
+    if (wrappingKey == null) {
+      throw new Error("No encryption key provided for encapsulation.");
+    }
+
+    return await this.encryptUint8Array(privateKeyPkcs8, wrappingKey);
+  }
+
+  async wrapEncapsulationKey(
+    encapsulationKeySpki: Uint8Array,
+    wrappingKey: SymmetricCryptoKey,
+  ): Promise<EncString> {
+    if (encapsulationKeySpki == null) {
+      throw new Error("No encapsulation key provided for encapsulation.");
+    }
+
+    if (wrappingKey == null) {
+      throw new Error("No encryption key provided for encapsulation.");
+    }
+
+    return await this.encryptUint8Array(encapsulationKeySpki, wrappingKey);
+  }
+
+  async wrapSymmetricKey(
+    encapsulatedKey: SymmetricCryptoKey,
+    key: SymmetricCryptoKey,
+  ): Promise<EncString> {
+    if (encapsulatedKey == null) {
+      throw new Error("No encapsulated key provided for encapsulation.");
+    }
+
+    if (key == null) {
+      throw new Error("No encryption key provided for encapsulation.");
+    }
+
+    return await this.encryptUint8Array(encapsulatedKey.key, key);
+  }
```

Comment:
```suggestion
  async wrapSymmetricKey(
    keyToBeWrapped: SymmetricCryptoKey,
    wrappingKey: SymmetricCryptoKey,
  ): Promise<EncString> {
    if (keyToBeWrapped == null) {
      throw new Error("No keyToBeWrapped provided for wrapping.");
    }

    if (wrappingKey == null) {
      throw new Error("No wrappingKey provided for wrapping.");
    }

    return await this.encryptUint8Array(keyToBeWrapped.key, wrappingKey);
  }
```

Match abstraction naming.